### PR TITLE
Closes #2138 Update messaging overview docs

### LINKED ISSUE #2138: Update messaging overview docs
I only updated the main code block in #2103, but while running through this with @jaketrookman, we realized there were other sections that needed updating

### PR DESCRIPTION
**Pull request by jaketrookman**
This PR (Closes #2138) updates the messaging documentation to remove the match the more current implementation of the server client communication and variable passing.

### CODE REVIEW

File: training/MESSAGING_OVERVIEW.md [hunk 1]
```diff
@@ -57,21 +57,19 @@ The three components of the `registerFunction` method are the `cmd` string, the
...
 ```chapel
-proc serverFunctionMsg(cmd: string, payload: string, argSize: int, st: borrowed SymTab): MsgTuple throws {}
+proc serverFunctionMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {}
 ```
 
 The parameters in this function each server a specific purpose:
 - `cmd` is the command string used to access this function. It is used mostly for debugging as the `cmd` string used to access the function will be captured in the logger messages.
-- `payload` is the JSON string of arguments passed from the client.
-- `argSize` lets Chapel know how many arguments are in the payload JSON so it can properly create a map using the JSON.
+- `msgArgs` is the recieved information that the client passes as a `repMsg`. The information about the number of arguements ia bundled and will be pared accordingly with the denoted passed types. Individual arguements will need to be extracted on the server side.
```
Comment (Ethan-DeBandi99) [inline training/MESSAGING_OVERVIEW.md:65]: `repMsg` is not entirely correct here. I would state that it contains parameters required for the message processing instead.

File: training/MESSAGING_OVERVIEW.md [hunk 2]
```diff
@@ -57,21 +57,19 @@ The three components of the `registerFunction` method are the `cmd` string, the
...
 - `st: borrowed SymTab` is a reference parameter to the Server's Symbol Table. The Symbol Table acts like Arkouda's database; storing all information related to a data structure that can then be referenced from the Client without having to pass large amounts of data back and forth.
 
-All messages sent to the server are in JSON format. Before any arguments sent from the client can be used, they must be parsed. To do this use the function `parseMessageArgs(payload, argSize)`. This will parse your message arguments into a Map that can then be accessed using accessor methods. The most common of these accessor methods is `getValueOf(argName)` which will return the value of an argument that matches the provided name.
+All messages sent to the server are in JSON format. Before any arguments sent from the client can be used, they must be parsed. Each message arguement that is sent to the server needs to be parsed into its own variable to be accessed, using a Map that can then be accessed using accessor methods. The most common of these accessor methods is `getValueOf(argName)` which will return the value of an argument that matches the provided name.
```
Comment (Ethan-DeBandi99) [inline training/MESSAGING_OVERVIEW.md:68]: The parsing now happens automatically, so I am not sure we want to indicate this to the user. I would state that the JSON sent to the server is used to generate a MessageArgs object. That object maps the parameter name to the value. The values are always strings, but functions are provided to access it as the correct type for processing.

File: training/MESSAGING_OVERVIEW.md [hunk 3]
```diff
@@ -57,21 +57,19 @@ The three components of the `registerFunction` method are the `cmd` string, the
...
 
-For example, assuming the server request in the above section is what is being received, there are two arguments in the JSON, `arg1` and `arg2`. To access these arguments in Chapel, we will parse the message, or `payload`, into a variable called `msgArgs`. Then using the `getValueOf()` accessor method, we will set variables `arg1` and `arg2` equal to the values of their corresponding JSON arguments.
+For example, assuming the server request in the above section is what is being received, there are two arguments in the JSON, `arg1` and `arg2`. To access these arguments in Chapel, we will parse the messageinto a variable called `msgArgs`. Then using the `getValueOf()` accessor method, we will set variables `arg1` and `arg2` equal to the values of their corresponding JSON arguments.
```
Comment (Ethan-DeBandi99) [inline training/MESSAGING_OVERVIEW.md:70]:
```suggestion
For example, assuming the server request in the above section is what is being received, there are two arguments in the JSON, `arg1` and `arg2`. To access these arguments in Chapel, we will parse the message into a variable called `msgArgs`. Then using the `getValueOf()` accessor method, we will set variables `arg1` and `arg2` equal to the values of their corresponding JSON arguments.

```

I think we also should highlight how `int` and other type args can be accessed as that type.

**Review by Ethan-DeBandi99 — CHANGES_REQUESTED** [review]
Few updates to make. I want @pierce314159 to read through as well because he is way better with explanations than I am.

File: training/MESSAGING_OVERVIEW.md [hunk 4]
```diff
@@ -57,21 +57,19 @@ The three components of the `registerFunction` method are the `cmd` string, the
...
 ```chapel
-proc serverFunctionMsg(cmd: string, payload: string, argSize: int, st: borrowed SymTab): MsgTuple throws {}
+proc serverFunctionMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {}
 ```
 
 The parameters in this function each server a specific purpose:
 - `cmd` is the command string used to access this function. It is used mostly for debugging as the `cmd` string used to access the function will be captured in the logger messages.
-- `payload` is the JSON string of arguments passed from the client.
-- `argSize` lets Chapel know how many arguments are in the payload JSON so it can properly create a map using the JSON.
+- `msgArgs` contains parameters required for the message processing. The information about the number of arguements is bundled and will be pared accordingly with the denoted passed types. Individual arguements will need to be extracted on the server side.
```
Comment (Ethan-DeBandi99) [inline training/MESSAGING_OVERVIEW.md:65]:
```suggestion
- `msgArgs` contains parameters required for the message processing. The information about the number of arguements is bundled and will be paired accordingly with the denoted passed types. Individual arguements will need to be extracted on the server side.

```

File: training/MESSAGING_OVERVIEW.md [hunk 5]
```diff
@@ -57,21 +57,19 @@ The three components of the `registerFunction` method are the `cmd` string, the
...
 
-All messages sent to the server are in JSON format. Before any arguments sent from the client can be used, they must be parsed. To do this use the function `parseMessageArgs(payload, argSize)`. This will parse your message arguments into a Map that can then be accessed using accessor methods. The most common of these accessor methods is `getValueOf(argName)` which will return the value of an argument that matches the provided name.
+All messages sent to the server are in JSON format. The JSON message sent to the server is used to generate a MessageArgs object and will be parsed automatically. These objects map the parameter name to the value using accessor methods. The values are always strings, but functions are provided to access it as the correct type for processing. The most common of these accessor methods is `getValueOf(argName)` which will return the value of an argument that matches the provided name.
```
Comment (Ethan-DeBandi99) [inline training/MESSAGING_OVERVIEW.md:68]:
```suggestion
All messages sent to the server are in JSON format. The JSON message sent to the server is used to generate a MessageArgs object automatically. These objects map the parameter name to the value using accessor methods. The values are always strings, but functions are provided to access it as the correct type for processing. The most common of these accessor methods is `getValueOf(argName)` which will return the value of an argument that matches the provided name.

```

File: training/MESSAGING_OVERVIEW.md [hunk 6]
```diff
@@ -57,21 +57,19 @@ The three components of the `registerFunction` method are the `cmd` string, the
...
 
-For example, assuming the server request in the above section is what is being received, there are two arguments in the JSON, `arg1` and `arg2`. To access these arguments in Chapel, we will parse the message, or `payload`, into a variable called `msgArgs`. Then using the `getValueOf()` accessor method, we will set variables `arg1` and `arg2` equal to the values of their corresponding JSON arguments.
+For example, assuming the server request in the above section is what is being received, there are two arguments in the JSON, `arg1` and `arg2`. To access these arguments in Chapel, we will parse the messageinto a variable called `msgArgs`. Then using the `getValueOf()` accessor method, we will set variables `arg1` and `arg2` equal to the values of their corresponding JSON arguments. These arguments can be accessed as their respective types, int and other type args.
```
Comment (Ethan-DeBandi99) [inline training/MESSAGING_OVERVIEW.md:70]:
```suggestion
For example, assuming the server request in the above section is what is being received, there are two arguments in the JSON, `arg1` and `arg2`. To access these arguments in Chapel, we will  use the `msgArgs` variable that is automatically generated. Then using the `getValueOf()` accessor method, we will set variables `arg1` and `arg2` equal to the values of their corresponding JSON arguments. These arguments can be accessed as their respective types, int and other type args.

```

**Review by Ethan-DeBandi99 — CHANGES_REQUESTED** [review]
Few small wording adjustments. Feel free to use my suggestions or retool as you see fit. I would like to point out we should highlight some of the type specific accessors because they are different than just calling `.getValueOf`.

File: training/MESSAGING_OVERVIEW.md (hunk 4 above)
Comment (pierce314159) [reply training/MESSAGING_OVERVIEW.md:65]: just adding on that there's a typo in `arguements`

File: training/MESSAGING_OVERVIEW.md (hunk 6 above)
Comment (pierce314159) [reply training/MESSAGING_OVERVIEW.md:70]: I think the last line should be reworded or removed. `int and other type args` just doesn't read write to me

**Review by pierce314159 — COMMENTED** [review]
Few small things but looks good otherwise!

File: training/MESSAGING_OVERVIEW.md (hunk 3 above)
Comment (Ethan-DeBandi99) [reply training/MESSAGING_OVERVIEW.md]: New suggestion made here. Resolving.

**Review by Ethan-DeBandi99 — CHANGES_REQUESTED** [review]
Need to update the example to correct a function call.

**Review by Ethan-DeBandi99 — APPROVED** [review]

File: training/MESSAGING_OVERVIEW.md [hunk 7]
```diff
@@ -57,23 +57,21 @@ The three components of the `registerFunction` method are the `cmd` string, the
...
 ```chapel
-proc serverFunctionMsg(cmd: string, payload: string, argSize: int, st: borrowed SymTab): MsgTuple throws {}
+proc serverFunctionMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {}
 ```
 
 The parameters in this function each server a specific purpose:
 - `cmd` is the command string used to access this function. It is used mostly for debugging as the `cmd` string used to access the function will be captured in the logger messages.
-- `payload` is the JSON string of arguments passed from the client.
-- `argSize` lets Chapel know how many arguments are in the payload JSON so it can properly create a map using the JSON.
+- `msgArgs` contains parameters required for the message processing. The information about the number of arguements is bundled and will be paired accordingly with the denoted passed types. Individual arguments will need to be extracted on the server side.
```
Comment (pierce314159) [inline training/MESSAGING_OVERVIEW.md:65]:
```suggestion
- `msgArgs` contains parameters required for the message processing. The information about the number of arguments is bundled and will be paired accordingly with the denoted passed types. Individual arguments will need to be extracted on the server side.
```

**Review by pierce314159 — APPROVED** [review]

**Pull request merged**